NitroxServer/GameLogic/Entities/Spawning/BatchEntitySpawner.cs

```diff
@@ -310,19 +318,35 @@ private IEnumerable<Entity> CreateEntityWithChildren(EntitySpawnPoint entitySpaw
             spawnedEntity.ChildEntities = SpawnEntities(entitySpawnPoint.Children, deterministicBatchGenerator, spawnedEntity);
         }
 
+        List<Entity> newCellRootChildren = [];
+
+        for (int i = spawnedEntity.ChildEntities.Count - 1; i >= 0; i--)
+        {
+            Entity spawnedChildEntity = spawnedEntity.ChildEntities[i];
+            // Regular case in which a spawned child would actually need to be spawned under the cell root instead of the current entity
+            if (spawnedChildEntity.ParentId != spawnedEntity.Id)
+            {
+                spawnedEntity.ChildEntities.RemoveAt(i);
+                newCellRootChildren.Add(spawnedChildEntity);
+            }
+        }
+
```

Comment:
Can this be moved after `spawnedEntities` init to remove the `newCellRootChildren` list?